integrations/tracer/opentelemetry/tracer_test.go

```
@@ -0,0 +1,109 @@
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatal("expect no panic")
+		}
+	}()
+
+	tracer := NewTracer("testing")
+
+	// NOTE: For now all type of context always return a noop span by otel.
+
+	t.Run("Nil", func(t *testing.T) {
+		span := tracer.SpanFromContext(nil)
```

golangci-lint (1.62.2)

SA1012: do not pass a nil Context, even if a function permits it; pass context.TODO if you are unsure about which Context to use

(staticcheck)